app/services/events/models.py

```diff
@@ -0,0 +1,116 @@
+from pydantic import BaseModel, Field, field_validator
+from datetime import datetime, date
+from typing import List, Optional
+
+
+class Event(BaseModel):
+    """Event model with validation."""
+
+    title: str = Field(..., min_length=1, description="Event title")
+    category: str = Field(..., min_length=1, description="Event category")
+    date: datetime = Field(..., description="Event date and time")
+    venue: str = Field(..., min_length=1, description="Event venue")
+    url: str = Field(..., min_length=1, description="Event URL")
+
+    @field_validator("title", "category", "venue")
+    @classmethod
+    def validate_non_empty_strings(cls, v):
+        """Ensure string fields are not empty after stripping."""
+        if not v or not v.strip():
+            raise ValueError("Field cannot be empty")
+        return v.strip()
+
+    @field_validator("url")
+    @classmethod
+    def validate_url(cls, v):
+        """Basic URL validation."""
+        if not v or not v.strip():
+            raise ValueError("URL cannot be empty")
+        v = v.strip()
+        if not (v.startswith("http://") or v.startswith("https://")):
+            raise ValueError("URL must start with http:// or https://")
+        return v
+
+
+class EventQuery(BaseModel):
+    """Simple model for parsing event search queries from user input."""
+
+    city: str = Field(..., description="Destination city")
+    start_date: Optional[date] = Field(
+        None,
+        description="Trip start date in ISO format (YYYY-MM-DD) if mentioned, else null",
```

Comment:
The field description refers to "YYYY-MM-DD" but the type is date, which may cause confusion if a string is supplied. Add a pre-validator to convert from string to date.
```
@field_validator("start_date", "end_date", mode="before")
@classmethod
def parse_date(cls, v):
    if isinstance(v, str) and v:
        return date.fromisoformat(v)
    return v
```